### DISCUSSION TITLE
The number of items in the submarine is calculated incorrectly in merchant's list of goods.

### DISCUSSION THREAD

**Discussion by pavel4win** [FakeFishGames/Barotrauma#13828] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

The number of items in the submarine is calculated incorrectly in merchant's list of goods.
On the screenshot you can see 2 thorium rods in the shop and 0 in the submarine. 

![2024-04-25_14-01-37](https://github.com/FakeFishGames/Barotrauma/assets/47295383/08bb1183-567a-4b7f-9589-272a9af6ef08)

But I know for sure that there are 2 thorium rods in the cabinet near the reactor.

![2024-04-25_14-03-22](https://github.com/FakeFishGames/Barotrauma/assets/47295383/edbf13f7-b0f1-4fa4-8eb5-d549122b7197)

When I moved them from the cabinet to the bot's inventory, the rods appeared in the merchant's list.

![2024-04-25_14-04-01](https://github.com/FakeFishGames/Barotrauma/assets/47295383/4385fea8-a755-4b73-a1ba-773d4d69f8c4)

![2024-04-25_14-04-38](https://github.com/FakeFishGames/Barotrauma/assets/47295383/f70aeed5-6f2b-4a5f-a777-8391be98985e)

It seems like items lying on the submarine in the cabinets are now not taken into account.

### Reproduction steps

Steps to reproduce described above

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.4.4.1 (Blood in the Water Update)

### Which operating system did you encounter this bug on?

Windows

**Comment by NilanthAnimosus** [comment]
This has been fixed in our private repo now and should be included in an upcoming hotfix!
